ui/imports/shared/views/AssetsView.qml

```diff
@@ -74,8 +74,8 @@ Control {
     signal swapRequested(string key)
     signal assetClicked(string key)
     signal communityClicked(string communityKey)
-    signal hideRequested(string key)
-    signal hideCommunityAssets(string communityKey)
+    signal hideRequested(string key, string assetSymbol, string assetName, string assetImage, bool isCommunityToken)
+    signal hideCommunityAssetsRequested(string communityKey)
```

Comment:
Thanks for removing calls to `Gobal` and preserving those signals.

However for consistency we should also externalize `ConfirmHideCommunityAssetsPopup`, along with `ConfirmHideAssetPopup`. We can put it just in `RightTabView` for now. Otherwise `hideRequested` and `hideCommunityAssetsRequested` are inconsistent as one is kind of unconditional and another one is intended to be confirmed externally.

Also `hideRequested` should stay as just `signal hideRequested(string key)` on the caller side we can use `ModelUtils.getByKey` to fetch whatever is needed for external component/action.
This point is covered in https://github.com/status-im/status-desktop/blob/73e86a37bc290116ef7a218bf6a6084a7d6e66e7/architecture-guide.md, with some examples and rationale (section starting with `Action signals (e.g. click on delegate) should not provide metadata...`.

Reply:
> Thanks for removing calls to `Gobal` and preserving those signals.
> 
> However for consistency we should also externalize `ConfirmHideCommunityAssetsPopup`, along with `ConfirmHideAssetPopup`. We can put it just in `RightTabView` for now. Otherwise `hideRequested` and `hideCommunityAssetsRequested` are inconsistent as one is kind of unconditional and another one is intended to be confirmed externally.

As explained before, I can't really move the `ConfirmHideAssetPopup` from `Popups` to `RightTabView` as another global popup (`FirstTokenReceivedPopup`) depends on it.

> Also `hideRequested` should stay as just `signal hideRequested(string key)` on the caller side we can use `ModelUtils.getByKey` to fetch whatever is needed for external component/action. This point is covered in https://github.com/status-im/status-desktop/blob/73e86a37bc290116ef7a218bf6a6084a7d6e66e7/architecture-guide.md, with some examples and rationale (section starting with `Action signals (e.g. click on delegate) should not provide metadata...`.

That assumes, on the caller's side, we have exactly the same model that triggered this popup, does that always hold true?

Reply:
> As explained before, I can't really move the `ConfirmHideAssetPopup` from `Popups` to `RightTabView` as another global popup (`FirstTokenReceivedPopup`) depends on it.

Yeah, sure, I'm proposing aligning in the opposite way. To consistently remove both popups from `AssetsView`, not touching `Popups`.

> That assumes, on the caller's side, we have exactly the same model that triggered this popup, does that always hold true?

It's not necessarily the same model (e.g. ui component may use sfpm internally). The key thing is that outside we always have the same information because we had to pass it to that UI component from that level.

Here in `AssetsView` (and in general), using that component as a proxy for popup params means tight coupling of that component and component (popup) handling that event. E.g. what if we decided that popup needs `description` of a token. The change is in popup, but we need to update `AssetsView` regarding what's expected in the model and amend signal signature as well. And it may be more complicated, we shouldn't assume that component requesting something will always provide all necessary details to handle that request.

Reply:
OK fixed both points; pls have another look :)